### PR TITLE
Sm/ignorable-metadata-resolver

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
option for metadataResolver to not use the local forceIgnore (only use its default)

This is needed for the WI, where source:status wants to show ignored files but is using SDR to resolve nam/type from filenames and they're always ignored.

side issue--I was getting some errors from running tests, and this remove some of the test reporter stuff that was erroring.

### What issues does this PR fix or reference?
@W-9805809@

### Functionality Before
local files are always ignored

### Functionality After
local files can be shown by setting a param to false

### CODE REVIEW

File: src/resolve/metadataResolver.ts [hunk 1]
```diff
@@ -49,7 +49,10 @@ export class MetadataResolver {
       throw new TypeInferenceError('error_path_not_found', fsPath);
     }
 
-    this.forceIgnore = ForceIgnore.findAndCreate(fsPath);
+    // use the default ignore if we aren't using a real one
+    this.forceIgnore = this.useRealForceIgnore
```
Comment (WillieRuemmele) [inline src/resolve/metadataResolver.ts:53]: NIT: I don't love `real` in the var name. Especially given the fake mock registry used in tests. What do you think about `findForceIgnore` or `useDefaultForceignore`?

File: src/resolve/metadataResolver.ts [hunk 2]
```diff
@@ -23,17 +23,17 @@ export class MetadataResolver {
...
    * @param registry Custom registry data
    * @param tree `TreeContainer` to traverse with
    */
-  constructor(registry = new RegistryAccess(), tree: TreeContainer = new NodeFSTreeContainer()) {
-    this.registry = registry;
-    this.tree = tree;
+  constructor(
+    private registry = new RegistryAccess(),
+    private tree: TreeContainer = new NodeFSTreeContainer(),
+    private useRealForceIgnore = true
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/resolve/metadataResolver.ts:35]: You don't want to pass in a forceIgnore.  `this.forceIgnore` is set each time in `getComponentsFromPath` and is relative to the path being passed into that.

At the class level, it's either
1. find the forceignore for that path
2. don't

Setting it in `getComponentsFromPath` only if it's not set previously would be quite different, I think.

[the remark this answers is not in the document]
Comment (mshanemc) [reply src/resolve/metadataResolver.ts:35]: Or...maybe that would be a good change, perf-wise, if you're resolving a lot of components?

[the remark this answers is not in the document]
Comment (mshanemc) [reply src/resolve/metadataResolver.ts:35]: This is another situation where someone deferred that step into the method instead initializing in the constructor, but I don't have comments to know why they might have done that.

**Pull request merged**